### ISSUE TITLE
#suggestion#many toolkit is used the epochs parameters to train machine translation model and t2t why not do that ?

### ISSUE THREAD

**Issue by zxqchat** [tensorflow/tensor2tensor#1041]
some times the loss is go down, but it's bleu score not always become better.especially，when I used the average model as Evaluation model,too many steps caused performence decline.

**Comment by martinpopel** [comment]
I am not sure what is your problem or question. You should choose the best checkpoint (early stopping to prevent overfitting) based on the most extrinsic and most reliable available metric. In your case it is probably the BLEU score on a dev set (which comes from the same domain as your test set).
